google-cloud-spanner/src/main/java/com/google/cloud/spanner/connection/UnitOfWork.java

```diff
@@ -39,9 +41,11 @@
 
   enum UnitOfWorkState {
     STARTED,
+    COMMITTING,
```

Comment:
State added as a `commitAsync` will initiate a commit, but return before the outcome of the commit is known. Setting this state is needed in order to know that the following statement should be executed in a new transaction.